recipe/meta.yaml

```diff
@@ -1,90 +1,57 @@
-  string: cpu_{{ blas_impl }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                 # [(osx and x86_64) or cuda_compiler_version == "None"]
-  string: mps_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                                 # [osx and arm64]
-  missing_dso_whitelist:
-    - "**/libcuda.so*"
- 
-  script:
-    - LLAMA_ARGS="-DLLAMA_BUILD_TESTS=OFF"              # [unix]
-    - set LLAMA_ARGS=-DLLAMA_BUILD_TESTS=OFF            # [win]
-    {% macro llama_args(value) -%}
-    - LLAMA_ARGS="${LLAMA_ARGS} -DLLAMA_{{ value }}"    # [unix]
-    - set LLAMA_ARGS=%LLAMA_ARGS% -DLLAMA_{{ value }}   # [win]
-    {%- endmacro %}
-    {% macro cmake_args(value) -%}
-    - LLAMA_ARGS="${LLAMA_ARGS} -D{{ value }}"          # [unix]
-    - set LLAMA_ARGS=%LLAMA_ARGS% -D{{ value }}         # [win]
-    {%- endmacro %}
-
-    {{ cmake_args("BUILD_SHARED_LIBS=ON") }}
-
-    {{ llama_args("NATIVE=OFF") }}      # [cpu_variant == "main_channel_default"]
-    {{ llama_args("AVX=OFF") }}         # [osx and arm64]
-    {{ llama_args("AVX2=OFF") }}        # [osx and arm64]
-    {{ llama_args("FMA=OFF") }}         # [osx and arm64]
-    {{ llama_args("F16C=OFF") }}        # [osx and arm64]
-    {{ llama_args("METAL=ON") }}        # [osx and arm64]
-    {{ llama_args("METAL=OFF") }}       # [osx and x86_64]
-    {{ llama_args("ACCELERATE=ON") }}   # [osx]
-
-    {{ llama_args("CUBLAS=ON") }}                     # [cuda_compiler_version != "None"]
-    {{ cmake_args("CMAKE_CUDA_ARCHITECTURES=all") }}  # [cuda_compiler_version != "None"]
-
-    {{ llama_args("BLAS=ON") }}                     # [not osx and cuda_compiler_version == "None"]
-    {{ llama_args("BLAS_VENDOR=Intel10_64_dyn") }}  # [not osx and cuda_compiler_version == "None" and blas_impl == "mkl"]
-
-    - echo $LLAMA_ARGS  # [unix]
-    - set LLAMA_ARGS    # [win]
-
-    - export CUDACXX=/usr/local/cuda/bin/nvcc   # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
-    - export CUDAHOSTCXX="${CXX}"               # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
-    - export LDFLAGS="$LDFLAGS -Wl,-rpath-link,/usr/lib64"    # [(cuda_compiler_version or "").startswith("12")]
-
-    - cmake -S . -B build -G Ninja %CMAKE_ARGS% %LLAMA_ARGS%    # [win]
-    - cmake -S . -B build -G Ninja ${CMAKE_ARGS} ${LLAMA_ARGS}  # [unix]
-    - cmake --build build
-    - cmake --install build 
-
+  # skip_gpu is set through abs.yaml for use in prefect only
+  skip: true # [skip_gpu and ((gpu_variant == "cuda-11") or (gpu_variant == "cuda-12"))]
+  skip: true # [s390x]
+  # Use a build number difference to ensure that the GPU
+  # variant is slightly preferred by conda's solver, so that it's preferentially
+  # installed where the platform supports it.
+  number: {{ build_number + 200 }}  # [gpu_variant == "cuda-12"]
```

Comment:
How about adding [this metapackage](https://github.com/AnacondaRecipes/pytorch-feedstock/blob/42463c4b5eb001eaf3f241d57ce2e5fd04735195/recipe/meta.yaml#L45) so that people can install the cpu variant easily if they need it?

Reply:
I find this pattern a bit cumbersome to maintain. Besides, people still have control through `conda search "llama.cpp=*=cpu_*"`

Reply:
>I find this pattern a bit cumbersome to maintain

yeah.. I think conda-forge might have a cpu or gpu metapackage or something which is better